Makefile

```diff
@@ -0,0 +1,131 @@
+# PREFIX is the prefix on the targetsystem
+# DESTDIR can be used to prefix ALL paths, e.g., to do a dummy-install in a
+# fake root dir, e.g., for building packages. Users mainly want PREFIX
+
+PREFIX=/usr/local
+DOCDIR=${PREFIX}/share/doc/skogul
+
+GIT_DESCRIBE:=$(shell git describe --always --tag --dirty)
+VERSION_NO=$(shell echo ${GIT_DESCRIBE} | sed s/[v-]//g)
+OS:=$(shell uname -s | tr A-Z a-z)
+ARCH:=$(shell uname -m)
+
+skogul: $(wildcard *.go */*.go */*/*.go)
+	@echo 🤸 go build !
+	@go build -ldflags "-X main.versionNo=$V" -o skogul ./cmd/skogul
+
+docs/skogul.rst: skogul
+	@echo 😽 Generating documentation $@
+	@./skogul -make-man > $@
+
+skogul.1: docs/skogul.rst
+	@echo 🎢 Generating man-file $@
+	@rst2man < $< > $@
+
+notes: docs/NEWS
+	@echo ⛲ Extracting release notes.
+	@./build/release-notes.sh ${GIT_DESCRIBE} > notes
+
+all: skogul skogul.1 docs/skogul.rst
+
+install: skogul skogul.1 docs/skogul.rst
+	@echo 🙅 Installing
+	@install -D -m 0755 skogul ${DESTDIR}${PREFIX}/bin/skogul
+	@install -D -m 0644 skogul.1 ${DESTDIR}${PREFIX}/share/man/man1/skogul.1
+	@install -D -m 0644 docs/examples/default.json ${DESTDIR}/etc/skogul/default.json
+	@cd docs; \
+	find . -type f -exec install -D -m 0644 {} ${DESTDIR}${DOCDIR}/{} \;
+	@install -D -m 0644 README.rst LICENSE -t ${DESTDIR}${DOCDIR}/
+
+
+# Any complaints on this macro-substitution without patches and I introduce m4.
+build/redhat-skogul.spec: build/redhat-skogul.spec.in
+	@echo  ❕Building spec-file
+	@cat $< | sed "s/xxVxx/${GIT_DESCRIBE}/g; s/xxARCHxx/${ARCH}/g; s/xxVERSION_NOxx/${VERSION_NO}/g" > $@
+	@which dpkg >/dev/null 2>&1 && { echo 🆒 Adding debian-workaround for rpm build; sed -i 's/^BuildReq/\#Debian hack, auto-commented out: BuildReq/g' $@; } || true
```

Comment:
Should this be a "non yum/rpm" hack instead of a debian specific hack? As in, should the hack be applied if the system does not have yum, rather than if it has dpkg? That might make it work better on my MacOS at least. :shipit:

Reply:
Who in the world uses non-debian stacks?

And the problem is: I actually _have_ yum and rpm on my debian-box. But I suppose we can check for /etc/redhat-version, that should be fairly reliable. More so than lsb_release at least.